File: doacao_app_java/src/main/java/schemas/Adotante.java
package schemas;

public class Adotante extends Pessoa {
    private String preferenciaAdocao;

    public String getPreferenciaAdocao() {
        return preferenciaAdocao;
    }

    public void setPreferenciaAdocao(String preferenciaAdocao) {
        this.preferenciaAdocao = preferenciaAdocao;
    }

    public Adotante(Integer id, String nome, String senha, String endereco, String email, String documento, String telefone, String preferenciaAdocao) {
        super(id, nome, email, documento, telefone, senha, endereco);
        this.preferenciaAdocao = preferenciaAdocao;
    }

    @Override
    public void enviarMensagem(String texto) {
        System.out.println("schemas.Mensagem para o adotante " + getNome() + ": " + texto);
    }
}

File: doacao_animal_python/schemas/adotante.py
from .pessoa import Pessoa


class Adotante(Pessoa):
    preferenciaAdocao: str
    
    def __init__(self, id: int, nome: str, email: str, documento: str, telefone: str, senha: str, endereco: str, preferenciaAdocao: str):
        super().__init__(id, nome, email, documento, telefone, senha, endereco)
        self.preferenciaAdocao = preferenciaAdocao
    
    def enviarMensagem(self, msg):
        return super().enviarMensagem(msg)